modules/phase_field/src/functions/FourierNoise.C

```diff
@@ -0,0 +1,111 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FourierNoise.h"
+#include "MooseRandom.h"
+#include "FEProblemBase.h"
+#include "libmesh/utility.h"
+
+registerMooseObject("PhaseFieldApp", FourierNoise);
+
+template <>
+InputParameters
+validParams<FourierNoise>()
+{
+  InputParameters params = validParams<Function>();
+  params.addClassDescription("Generate noise from a fourier series");
+  params.addRequiredParam<Real>("lambda",
+                                "Wavelength cut off (set to about twice the interfacial width)");
+  params.addParam<unsigned int>(
+      "num_terms",
+      "Number of random fourier series terms (this will result in non-periodic noise). Omit this "
+      "parameter to obtain a periodic noise distribution.");
+  params.addParam<unsigned int>("seed", 12455, "Random number generator seed");
+  return params;
+}
+
+FourierNoise::FourierNoise(const InputParameters & parameters)
+  : Function(parameters),
+    _lambda(getParam<Real>("lambda")),
+    _fe_problem(*getCheckedPointerParam<FEProblemBase *>("_fe_problem_base"))
+{
+  MooseRandom::seed(getParam<unsigned int>("seed"));
+
+  if (isParamValid("num_terms"))
+  {
+    // random terms
+    _series.resize(getParam<unsigned int>("num_terms"));
+    const Real scale = 2.0 * (2.0 * libMesh::pi) / _lambda;
+
+    // check
+    if (_series.empty())
+      paramError("num_terms",
+                 "If specifying the number of terms, supply a number greater than zero.");
+
+    // fill terms
+    for (auto & f : _series)
+    {
+      // get a vector with length <= 0.5
+      Real r2;
+      do
+      {
+        const Real x = MooseRandom::rand() - 0.5;
```

Comment:
Why are you using the static instance here? I thought you wanted to exterminate it. I would suggest using a local instance so that the inclusion of this object doesn't change behaviors of other objects that still use the static version.

Reply:
That local instance is just so kludgy. The hash map is superfluous when you can just create multiple instances of MooseRandom...

Reply:
OK - so the interface isn't perfect, I agree. Perhaps we need an overload to just use some default state in MooseRandom that isn't backed up or restored (e.g. it behaves just like the static instance except it's not global. Would that satisfy your needs? I'd like to see the use of the static generator be discouraged. We have code all over the framework and module that interact with the static instance. Each one of them feeding seeds and getting out random numbers. In some cases we reseed regularly, which is kind of dangerous to your desire for good random numbers. As you add or remove these objects from your sim, you impact behavior, which I think is possibly worse than using a slightly clunky interface.

TL;DR; Let's fix up the interface to MooseRandom and avoid the static usage unless you are diametrically opposed.

Reply:
Also - there's this: https://github.com/idaholab/moose/issues/5567

Reply:
Can you have an overload with the same function signatures except one is `static` and the other one isn't?!

Reply:
No, directly prohibited by the standard:

https://stackoverflow.com/questions/5365689/c-overload-static-function-with-non-static-function

Reply:
Mys suggestion would be to remove static usage and keep the function signatures without the index parameter to refer to a separate state that is not stored in the map. It'll be a bit of a ripping-off-the-bandaid-pain to regold some tests, and fix some apps, but worth it IMO.

Reply:
I agree with that. I'm not sure I want to completely remove the static usage though. Perhaps we rename the static methods and maybe even print a warning when they are used? The static methods are still better than using system methods, but only just barely.

Reply:
nonetheless, this is all orthogonal to this PR